driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -789,6 +793,15 @@ private void setFinalResult(Connection connection, Message.Response response) {
         }
     }
 
+    private void logIdempotenceWarning() {
+        if (WARNED_IDEMPOTENT.compareAndSet(false, true))
```

Comment:
What do you think about establishing a common logger implementation for messages like this?  This way users can easily suppress warnings like this globally for all changes.   Some users pay very close attention to warnings and try their best to suppress all the ones that doesn't affect their application health.